pkg/saga/statemachine/engine/repo/repository/state_machine_repository.go

```diff
@@ -71,6 +71,14 @@ func GetStateMachineRepositoryImpl() *StateMachineRepositoryImpl {
 	return stateMachineRepositoryImpl
 }
 
+func (s *StateMachineRepositoryImpl) SetDefaultTenantId(tenantId string) {
+	s.defaultTenantId = tenantId
+}
+
+func (s *StateMachineRepositoryImpl) GetDefaultTenantId() string {
+	return s.defaultTenantId
```

Comment:
pkg/saga/statemachine/engine/config/default_statemachine_config.go
pkg/saga/statemachine/engine/repo/repository/state_machine_repository.go
Issue

The tenant field is only set in some registration/loading locations. State instances, compensation chains, logs, and other functions do not fully carry tenant information, which may lead to cross-tenancy.
Suggestion

Add the TenantId field to structures such as StateMachineInstance and StateInstance. All persistence, logs, and compensation chains carry tenant information.
All methods, such as GetStateMachineByNameAndTenantId and RegistryStateMachine, must pass and verify tenant information.